### DISCUSSION TITLE
Configure GTK file chooser dialog for downloads

### DISCUSSION THREAD

**Discussion by cmvanb** [qutebrowser/qutebrowser#8058] [Q&A]
What I want to achieve is to use the GTK file dialog when choosing a path to download files, instead of the builtin dialog.

I have configured my wayland system with xdg-desktop-portal-wlr and xdg-desktop-portal-gtk.  This works in chromium when uploading or saving a file from a web site (e.g. `https://app.diagrams.net`).

In qutebrowser it seems to use a different dialog (probably QT?). 

I tried using the setting `c.fileselect.handler = 'external'`, but since the GTK file chooser is activated through xdg desktop portal, I'm lost as to how to configure the command for `c.fileselect.single_file.command`, etc.

Is it possible to configure qutebrowser to use the GTK file picker?

EDIT FOR _PARTIAL_ SOLUTION:

I was able to get this working:
```
c.fileselect.handler = 'external'
c.fileselect.single_file.command = [ 'yad', '--file', '--width', '1280', '--height', '800' ]
c.fileselect.multiple_files.command = [ 'yad', '--file', '--width', '1280', '--height', '800'  ]
c.fileselect.folder.command = [ 'yad', '--file', '--directory', '--width', '1280', '--height', '800'  ]
```

[yad](https://github.com/v1cont/yad) is a command line program for spawning GTK dialogs, including options for a file/directory selector. This works in Qutebrowser for uploading/saving a file.

It also works with the download prompt, although I have not figured out how I can configure it to use the external fileselect handler by default, rather than having to press `Alt+e` every time...

**Comment by The-Compiler** [comment]
I reopened #1365 for this. I don't think we currently have a way to always automatically spawn the external file selector for downloads.

**Comment by isthatelitho** [comment]
Hey, just wanted to mention that zenity works way better than yad for this:

```
c.fileselect.handler = 'external'

c.fileselect.single_file.command = ['zenity', '--file-selection', '--title=Select File']
c.fileselect.multiple_files.command = ['zenity', '--file-selection', '--multiple', '--title=Select Files']
c.fileselect.folder.command = ['zenity', '--file-selection', '--directory', '--title=Select Folder']
```

Zenity uses the actual native GTK file chooser widget (same one Firefox uses), so you get all the modern features and Yad just creates a basic custom dialog that looks pretty dated in comparison.
